### DISCUSSION TITLE
Blockcheck для проверки стратегии антизамедления? Интерпретация результатов блокчека и курла

### DISCUSSION THREAD

**Discussion by muzzy124** [bol-van/zapret#272] [General]
Подходит ли принципиально блокчек для проверки антизамедления?

Результат например на гуглвидео:

* SUMMARY
ipv4 googlevideo.com curl_test_http : tpws --hostdot
ipv4 googlevideo.com curl_test_http : nfqws not working
ipv4 googlevideo.com curl_test_https_tls12 : tpws not working
ipv4 googlevideo.com curl_test_https_tls12 : nfqws not working
ipv4 googlevideo.com curl_test_https_tls13 : tpws not working
ipv4 googlevideo.com curl_test_https_tls13 : nfqws not working

Но при этом стандартная стратегия nfqws из примера (MODE=nfqws, NFQWS_OPT_DESYNC="--dpi-desync=split2",
NFQWS_OPT_DESYNC_QUIC="--dpi-desync=fake --dpi-desync-repeats=6", MODE_HTTPS=1, MODE_QUIC=1)
работает обычно хорошо для ПК и андроид устройств на моем провайдере, но не работает для тизена/вебос, частично для ios.
Собственно пытался и блокчеком и курлом прогонять разные способы, ни один не работает для гуглвидео из командной строки или блокчека.

Пример (со nfqws стратегией выше):
```
$ curl --http3-only -v4s -o/dev/null -k --connect-to ::google.com -k -H Host:\ metsalehti-staging-s4uzwwd6nq-lz.a.run.app https://test.googlevideo.com/app/uploads/2021/11/2022-mediakortti.pdf -w %{speed_download}
* Connecting to hostname: google.com
* Host google.com:443 was resolved.
* IPv6: (none)
* IPv4: 142.250.186.46
*   Trying 142.250.186.46:443...
* Server certificate:
*  subject: CN=*.google.com
*  start date: Jul 30 12:32:53 2024 GMT
*  expire date: Oct 22 12:32:52 2024 GMT
*  issuer: C=US; O=Google Trust Services; CN=WR2
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
*   Certificate level 0: Public key type EC/prime256v1 (256/128 Bits/secBits), signed using sha256WithRSAEncryption
*   Certificate level 1: Public key type RSA (2048/112 Bits/secBits), signed using sha256WithRSAEncryption
*   Certificate level 2: Public key type RSA (4096/152 Bits/secBits), signed using sha256WithRSAEncryption
* Connected to google.com (142.250.186.46) port 443
* using HTTP/3
* [HTTP/3] [0] OPENED stream for https://test.googlevideo.com/app/uploads/2021/11/2022-mediakortti.pdf
* [HTTP/3] [0] [:method: GET]
* [HTTP/3] [0] [:scheme: https]
* [HTTP/3] [0] [:authority: metsalehti-staging-s4uzwwd6nq-lz.a.run.app]
* [HTTP/3] [0] [:path: /app/uploads/2021/11/2022-mediakortti.pdf]
* [HTTP/3] [0] [user-agent: curl/8.9.0]
* [HTTP/3] [0] [accept: */*]
> GET /app/uploads/2021/11/2022-mediakortti.pdf HTTP/3
> Host: metsalehti-staging-s4uzwwd6nq-lz.a.run.app
> User-Agent: curl/8.9.0
> Accept: */*
>
* Request completely sent off
< HTTP/3 200
< content-type: application/pdf
< last-modified: Mon, 10 Jan 2022 15:11:19 GMT
< vary: Accept-Encoding
< etag: "61dc4c97-6d4968"
< x-frame-options: SAMEORIGIN
< accept-ranges: bytes
< x-cloud-trace-context: db60e93481faa9ef2c3c88bb797f2aaf;o=1
< date: Wed, 14 Aug 2024 06:03:54 GMT
< server: Google Frontend
< content-length: 7162216
< alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
<
{ [1170 bytes data]
* Connection #0 to host google.com left intact
4770914

```
```
curl  -v4s -o/dev/null -k --connect-to ::google.com -k -H Host:\ metsalehti-staging-s4uzwwd6nq-lz.a.run.app https://test.googlevideo.com/app/uploads/2021/11/2022-mediakortti.pdf -w %{speed_download}
* Connecting to hostname: google.com
* Host google.com:443 was resolved.
* IPv6: (none)
* IPv4: 142.250.185.238
*   Trying 142.250.185.238:443...
* Connected to google.com (142.250.185.238) port 443
* ALPN: curl offers http/1.1
} [5 bytes data]
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
} [512 bytes data]
* TLSv1.3 (IN), TLS handshake, Server hello (2):
{ [122 bytes data]
```
и тишина.

Как это интерпретировать?

**Comment by z0mb1e-kgd** [comment]
`blockcheck.sh` работает только для полностью заблокированных сайтов (например, rutracker.org), для googlevideo.com он работать не будет. Стоит прогнать `blockcheck.sh` для дефолтного rutracker.org и выбрать рабочую стратегию. Универсального рецепта нет, очень зависит от провайдера и пути следования трафика. В моем случае помогло сменить NFQWS_OPT_DESYNC на `NFQWS_OPT_DESYNC="--dpi-desync=fake,disorder2 --dpi-desync-fooling=md5sig"`.

**Comment by numbereleven-a** [comment]


**Reply by bol-van** [reply]
у вас случаем не работает запрет во время блокчека ?
его надо останавлиавать

мсс это жесткий хак. применять только точечно когда других вариантов нет

**Comment by Wertas03** [comment]


**Reply by numbereleven-a** [reply]
> Подскажите, есть синтаксис у этих параметров какой-то и чтобы я логику не сломал? После Blockcheck [blockcheck.txt](https://github.com/user-attachments/files/17510645/blockcheck.txt) я пробовал результаты и у меня либо работает Ютуб, либо, всё, что запрещено, но без Ютуба.
> 
> Всё (инста, ФБ, BBC, трекеры все и т.д.), кроме Ютуба. и так работает: NFQWS_OPT_DESYNC="--dpi-desync=fake --dpi-desync-ttl=6" и так работает: NFQWS_OPT_DESYNC="--dpi-desync=fake,disorder2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig"
> 
> Ютуб, но без остального (ну BBC еще грузится, но без некоторых картинок) #NFQWS_OPT_DESYNC="--dpi-desync=split --dpi-desync-fooling=badseq" #NFQWS_OPT_DESYNC="--dpi-desync=fake,split --dpi-desync-ttl=5 --dpi-desync-fake-tls=0x00000000 --dpi-desync-split-pos=1" #NFQWS_OPT_DESYNC="--dpi-desync=fake --dpi-desync-ttl=2 --dpi-desync-fake-http=0x00000000"
> 
> Как запихнуть (и возможно ли) в одну строку параметры для Ютуба и остального.

через МНОЖЕСТВЕННЫЕ СТРАТЕГИИ --new, читать ридми

**Reply by M-A-M-B-A** [reply]
В связи с отличающийся блокировкой для разных GGC прошу рассмотреть возможность работы blockcheck по нескольким адресам с суммированием результатов проверки. Спасибо. 
Например:
`./blockcheck.sh rr2---sn-pivhx-n8vl.googlevideo.com, rr7---sn-pivhx-n8vs.googlevideo.com, rr1---sn-pivhx-n8vz.googlevideo.com`

```
* SUMMARY
ipv4 rr2---sn-pivhx-n8vl.googlevideo.com, curl_test_https_tls12 : test aborted, no reason to continue. curl code 3: the URL was not properly formatted
ipv4 rr7---sn-pivhx-n8vs.googlevideo.com, curl_test_https_tls12 : test aborted, no reason to continue. curl code 3: the URL was not properly formatted
ipv4 rr1---sn-pivhx-n8vz.googlevideo.com curl_test_https_tls12 : nfqws --dpi-desync=fake,split --dpi-desync-ttl=3 --dpi-desync-fake-tls=0x00000000
```

**Reply by NOFEXtreme** [reply]
> В связи с отличающийся блокировкой для разных GGC прошу рассмотреть возможность работы blockcheck по нескольким адресам с суммированием результатов проверки. Спасибо. Например: `./blockcheck.sh rr2---sn-pivhx-n8vl.googlevideo.com, rr7---sn-pivhx-n8vs.googlevideo.com, rr1---sn-pivhx-n8vz.googlevideo.com`
> 
> ```
> * SUMMARY
> ipv4 rr2---sn-pivhx-n8vl.googlevideo.com, curl_test_https_tls12 : test aborted, no reason to continue. curl code 3: the URL was not properly formatted
> ipv4 rr7---sn-pivhx-n8vs.googlevideo.com, curl_test_https_tls12 : test aborted, no reason to continue. curl code 3: the URL was not properly formatted
> ipv4 rr1---sn-pivhx-n8vz.googlevideo.com curl_test_https_tls12 : nfqws --dpi-desync=fake,split --dpi-desync-ttl=3 --dpi-desync-fake-tls=0x00000000
> ```
Он так и работает. Домены разделяются пробелами, без запятых.

**Reply by M-A-M-B-A** [reply]
> > В связи с отличающийся блокировкой для разных GGC прошу рассмотреть возможность работы blockcheck по нескольким адресам с суммированием результатов проверки. Спасибо. Например: `./blockcheck.sh rr2---sn-pivhx-n8vl.googlevideo.com, rr7---sn-pivhx-n8vs.googlevideo.com, rr1---sn-pivhx-n8vz.googlevideo.com`
> > ```
> > * SUMMARY
> > ipv4 rr2---sn-pivhx-n8vl.googlevideo.com, curl_test_https_tls12 : test aborted, no reason to continue. curl code 3: the URL was not properly formatted
> > ipv4 rr7---sn-pivhx-n8vs.googlevideo.com, curl_test_https_tls12 : test aborted, no reason to continue. curl code 3: the URL was not properly formatted
> > ipv4 rr1---sn-pivhx-n8vz.googlevideo.com curl_test_https_tls12 : nfqws --dpi-desync=fake,split --dpi-desync-ttl=3 --dpi-desync-fake-tls=0x00000000
> > ```
> 
> Он так и работает. Домены разделяются пробелами, без запятых.

Да уже проверил. Через пробел работает.
